### DISCUSSION TITLE
Lab3 incorrect port declaration in fir.v file

### DISCUSSION THREAD

**Discussion by DanielDu9453** [bol-edu/HLS-SOC-Discussions#76] [Ideas]
I think that the port declarations for ss_tvalid and ss_tready are wrong.
ss_tvalid should be an input port of the fir module and ss_tready should be its output port.

Please correct me if this statement is wrong

**Comment by changde0628** [comment]
I also have the same problem. When I use iverilog, ss_tvalid will generate the following error message as output.
```
tb/fir_tb.v:65: error: reg ss_tvalid; cannot be driven by primitives or continuous assignment.
tb/fir_tb.v:65: error: Output port expression must support continuous assignment.
tb/fir_tb.v:65:      : Port 13 (ss_tvalid) of fir is connected to ss_tvalid
2 error(s) during elaboration.
``` 
And when I change ss_tvalid to output, I can compile it.
But I'm not sure if it's reasonable to do so.

**Comment by ZheChen-Bill** [comment]
同學你好，原先的版本有誤植，目前已經有修正ss_tvalid和ss_tready的I/O了，另外我們可能會微調一些關於SRAM的設計以及I/O，可能要麻煩你們重新確認，造成困擾十分抱歉。
